dubbo-common/src/main/java/org/apache/dubbo/common/metrics/model/ServiceInfo.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.common.metrics.model;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ServiceInfo {
+
+    private static final Map<String, ServiceInfo> caches = new ConcurrentHashMap<>();
+    private static final int maxSize = 200;
+
+    private String interfaceName;
+    private String methodName;
+    private String group;
+    private String version;
+
+    private ServiceInfo(String interfaceName, String methodName, String group, String version) {
+        this.interfaceName = interfaceName;
+        this.methodName = methodName;
+        this.group = group;
+        this.version = version;
+    }
+
+    public static ServiceInfo getInstance(String interfaceName, String methodName, String group, String version) {
```

Comment:
Can we just simply use `Invocation` here?

Reply:
Looking at the class description that Invocation is non-thread-safe, I'm not sure if the property will be modified by another unknown thread during use. Another developer, seeing that MethodMetric relies on a Invocation, might not immediately guess what the relationship is?